lametro/views.py

```diff
@@ -734,6 +736,100 @@ class LAMetroArchiveSearch(TemplateView):
 class LAMetroContactView(IndexView):
     template_name = 'lametro/contact.html'
 
+
+class MinutesView(EventsView):
+    template_name = 'lametro/minutes.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        start_date_str = None
+        end_date_str = None
+        if 'minutes-from' in self.request.GET.keys():
+            start_date_str = self.request.GET.get('minutes-from')
+            start_datetime = datetime.datetime.strptime(start_date_str, '%m/%d/%Y')
+        if 'minutes-to' in self.request.GET.keys():
+            end_date_str = self.request.GET.get('minutes-to')
+            end_datetime = datetime.datetime.strptime(end_date_str, '%m/%d/%Y')
+
+        # filter db events based on start and end dates
+        if end_date_str:
+            context['end_date'] = end_date_str
+            end_date_time = parser.parse(end_date_str)
+            stored_events = LAMetroEvent.objects.with_media()\
+                                                .filter(start_time__lt=end_date_time)\
+                                                .order_by('start_time')
+        else:
+            stored_events = LAMetroEvent.objects.with_media()\
+                                                .filter(start_time__lt=timezone.now())\
+                                                .order_by('start_time')
+        if start_date_str:
+            context['start_date'] = start_date_str
+            start_date_time = parser.parse(start_date_str)
+            stored_events = stored_events.filter(start_time__gt=start_date_time)\
+                                                .order_by('start_time')
+
+        filtered_db_events = []
+        for event in stored_events:
+            stored_events_dict = {
+                'start_time': event.start_time.date(),
+                'meeting': event.name,
+                'minutes_link': [],
+                'agenda_link': [],
+            }
+            try:
+                minutes_link = event.documents.get(note__icontains='minutes').links.first().url
+                stored_events_dict['minutes_link'].append(minutes_link)
+            except EventDocument.DoesNotExist:
+                pass
+
+            try:
+                agenda_link = event.documents.get(note__icontains='agenda').links.first().url
+                stored_events_dict['agenda_link'].append(agenda_link)
+            except EventDocument.DoesNotExist:
+                pass
+
+            filtered_db_events.append(stored_events_dict)
+
+        # filter events from csv based on start and end dates
+        csv_events = get_list_from_csv('historical_events.csv')
+        events_dicts = [dict(e) for e in csv_events]
```

Comment:
What is `e` before you cast it to a `dict`?

Reply:
it's an ordered dict, so it looks like a list of 2 tuples with what end up becoming key/value pairs

Reply:
Got it. While its representation in the console looks different, an OrderedDict [behaves just like a regular dictionary](https://docs.python.org/3/library/collections.html#collections.OrderedDict), only its order is guaranteed (vs. a standard dict, which is unordered) and it has a few additional methods to manipulate the order. So you don't need to cast `e` to dict here!